src/main/java/teammates/ui/webapi/action/UpdateInstructorPrivilegeAction.java

```diff
@@ -0,0 +1,154 @@
+            instructorToUpdate.privileges.updatePrivilege(
+                    sectionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS,
+                    request.isCanViewStudentInSections());
+
+            // updates session level privileges
+            if (sessionName == null) {
+                instructorToUpdate.privileges.updatePrivilege(
+                        sectionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS,
+                        request.isCanSubmitSessionInSections());
+                instructorToUpdate.privileges.updatePrivilege(
+                        sectionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS,
+                        request.isCanViewSessionInSections());
+                instructorToUpdate.privileges.updatePrivilege(
+                        sectionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION_COMMENT_IN_SECTIONS,
+                        request.isCanModifySessionCommentsInSections());
+            } else {
+                instructorToUpdate.privileges.updatePrivilege(sectionName, sessionName,
+                        Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS,
+                        request.isCanSubmitSessionInSections());
+                instructorToUpdate.privileges.updatePrivilege(sectionName, sessionName,
+                        Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS,
+                        request.isCanViewSessionInSections());
+                instructorToUpdate.privileges.updatePrivilege(sectionName, sessionName,
+                        Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION_COMMENT_IN_SECTIONS,
+                        request.isCanModifySessionCommentsInSections());
+            }
+        }
+
+        instructorToUpdate.privileges.validatePrivileges();
+        updateToEnsureValidityOfInstructorsForTheCourse(courseId, instructorToUpdate);
+
+        try {
+            logic.updateInstructor(
+                    InstructorAttributes
+                            .updateOptionsWithEmailBuilder(instructorToUpdate.courseId, instructorToUpdate.getEmail())
+                            .withPrivileges(instructorToUpdate.privileges)
+                            .build());
+        } catch (InvalidParametersException e) {
+            return new JsonResult(e.getMessage(), HttpStatus.SC_BAD_REQUEST);
+        } catch (EntityDoesNotExistException ednee) {
+            return new JsonResult(ednee.getMessage(), HttpStatus.SC_NOT_FOUND);
+        }
+
+        return new JsonResult("The instructor " + instructorToUpdate.getName()
+                + "'s privilege has been updated.", HttpStatus.SC_OK);
+    }
+
+
+    /**
+     * Checks if there are any other registered instructors that can modify instructors.
+     * If there are none, the instructor currently being edited will be granted the privilege
+     * of modifying instructors automatically.
```

Comment:
Is this V6's specification or you just add by yourself?

Reply:
Yes I get it from `UpdateInstructorAction` since I feel it is also applicable here.

Reply:
Sure. Let's keep it and make sure the new API can replace the privilege part of `UpdateInstructorAction`.